source/structured-streaming.txt

```diff
@@ -0,0 +1,327 @@
+MongoDB:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 40
+
+   * - Setting
+     - Description
+
+   * - ``.writeStream.format()``
+     - The format to use for write stream data. Use "mongodb".
+
+   * - ``.writeStream.option()``
+     - Use the ``.option`` method to specify your MongoDB deployment 
+       connection string with the "spark.mongodb.connection.uri" 
+       option key.
+
+       You must specify a database and collection, either as part of 
+       your connection string or with additional ``.option`` 
+       methods using the following keys:
+
+       - "spark.mongodb.database"
+       - "spark.mongodb.collection"
+
+   * - ``.writeStream.outputMode()``
+     - The output mode to use.
+
+.. tabs-drivers::
+
+   tabs:
+     - id: java-sync
+       content: |
+
+     - id: python
+       content: |
+
+         .. code-block:: python
+            :copyable: true
+            :emphasize-lines: 3-4, 7
+         
+            <streaming Dataset/ DataFrame> \
+              .writeStream \
+              .format("mongodb") \
+              .option("spark.mongodb.connection.uri", <mongodb-connection-string>) \
+              .option("spark.mongodb.database", <database-name>) \
+              .option("spark.mongodb.collection", <collection-name>) \
+              .outputMode("append")
+
+         For a complete list of methods, see the 
+         `pyspark Structured Streaming reference <https://spark.apache.org/docs/latest/api/python/reference/pyspark.ss.html>`__.
+
+     - id: scala
+       content: |
+
+.. _read-structured-stream:
+.. _continuous-processing:
+
+Configuring a Read Stream from MongoDB
+--------------------------------------
+
+Reading a stream from a MongoDB database requires 
+*continuous processing*, 
+an experimental feature introduced in Spark version 2.3. To learn 
+more about continuous processing, see the `Spark documentation <https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#continuous-processing>`__.
+
+To use continuous processing with the MongoDB Spark Connector, add the 
+``.trigger()`` method to the ``.writeStream`` of the streaming 
+Dataset or DataFrame that you create from your MongoDB ``.readStream``. 
+In your ``.trigger()``, specify the ``continuous`` parameter.
+
+.. note:: 
+
+   The connector populates its read stream from your MongoDB 
+   deployment's change stream. To populate your change stream, perform 
+   update operations on your database.
+
+   To learn more about change streams, see 
+   :manual:`Change Streams </changeStreams>` in the MongoDB manual.
+
+Specify read stream configuration settings on your local 
+SparkSession ``.readStream``. Lastly, add ``.load()`` to create a read 
+stream with your configuration.
+
+.. note:: 
+   
+   Spark does not begin streaming until you call the ``.start()`` 
+   method on a streaming query.
+
+You must specify the following configuration settings to read from 
+MongoDB:
```

Comment:
Suggestion: I think the note can be moved after the code snippet. This places the note closer to where `start` is called and doesn't break up these two connected sentences.

Suggestion: I think the sentence about `load` can be deleted or moved into the note about start as it is a Spark specific method that (AFAICT) is not unique to MongoDB.

```suggestion
Specify read stream configuration settings on your local 
SparkSession ``readStream`` property. You must specify the
following configuration settings to read from MongoDB:
```